src/allotropy/parsers/moldev_softmax_pro/softmax_pro_parser.py

```diff
@@ -1,17 +1,266 @@
+
+    def _get_model(self, file_name: str, data: Data) -> Model:
+        return Model(
+            field_asm_manifest="http://purl.allotrope.org/json-schemas/adm/plate-reader/BENCHLING/2023/09/plate-reader.schema",
+            plate_reader_aggregate_document=PlateReaderAggregateDocument(
+                device_system_document=DeviceSystemDocument(
+                    device_identifier="null",
+                    model_number="null",
+                ),
+                data_system_document=DataSystemDocument(
+                    file_name=file_name,
+                    software_name="SoftMax Pro",
+                    software_version=None,
+                    ASM_converter_name=ASM_CONVERTER_NAME,
+                    ASM_converter_version=ASM_CONVERTER_VERSION,
+                ),
+                plate_reader_document=[
+                    self._get_plate_reader_document_item(plate_block, position)
+                    for plate_block in data.get_plate_block()
+                    for position in plate_block.iter_wells()
+                ],
+                calculated_data_aggregate_document=None,
+            ),
+        )
+
+    def _get_plate_reader_document_item(
+        self, plate_block: PlateBlock, position: str
+    ) -> PlateReaderDocumentItem:
+        if plate_block.plate_block_type == "Absorbance":
+            return PlateReaderDocumentItem(
+                measurement_aggregate_document=MeasurementAggregateDocument(
+                    measurement_time=EPOCH,
+                    plate_well_count=TQuantityValueNumber(plate_block.header.num_wells),
+                    container_type=ContainerType.well_plate,
+                    measurement_document=list(
+                        self._iter_absorbance_measurement_document(
+                            plate_block, position
+                        ),
+                    ),
+                )
+            )
+        elif plate_block.plate_block_type == "Luminescence":
+            return PlateReaderDocumentItem(
+                measurement_aggregate_document=MeasurementAggregateDocument(
+                    measurement_time=EPOCH,
+                    plate_well_count=TQuantityValueNumber(plate_block.header.num_wells),
+                    container_type=ContainerType.well_plate,
+                    measurement_document=list(
+                        self._iter_luminescence_measurement_document(
+                            plate_block, position
+                        ),
+                    ),
+                )
+            )
+        elif plate_block.plate_block_type == "Fluorescence":
+            return PlateReaderDocumentItem(
+                measurement_aggregate_document=MeasurementAggregateDocument(
+                    measurement_time=EPOCH,
+                    plate_well_count=TQuantityValueNumber(plate_block.header.num_wells),
+                    container_type=ContainerType.well_plate,
+                    measurement_document=list(
```

Comment:
I don't see the point in making the helper methods return `Iterator`s if the only places they're called will immediately create lists from the iterators. How about just making the helpers return lists?